packages/data-stores/src/site/reducer.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { Reducer } from 'redux';
+import { combineReducers } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ActionType, NewSite, NewSiteErrorResponse } from './types';
+import * as Actions from './actions';
+
+const newSiteData: Reducer<
+	NewSite | {} | undefined,
+	| ReturnType< typeof Actions[ 'receiveNewSite' ] >
+	| ReturnType< typeof Actions[ 'receiveNewSiteFailed' ] >
+> = ( state = undefined, action ) => {
+	if ( action.type === ActionType.RECEIVE_NEW_SITE ) {
+		const { response } = action;
+		return {
+			...response.blog_details,
+		};
+	} else if ( action.type === ActionType.RECEIVE_NEW_SITE_FAILED ) {
+		return undefined;
+	}
+	return state;
+};
+
+const newSiteError: Reducer<
+	NewSiteErrorResponse | undefined,
+	| ReturnType< typeof Actions[ 'fetchNewSite' ] >
+	| ReturnType< typeof Actions[ 'receiveNewSite' ] >
+	| ReturnType< typeof Actions[ 'receiveNewSiteFailed' ] >
+> = ( state = undefined, action ) => {
+	switch ( action.type ) {
+		case ActionType.FETCH_NEW_SITE:
+			return undefined;
+		case ActionType.RECEIVE_NEW_SITE:
+			return undefined;
+		case ActionType.RECEIVE_NEW_SITE_FAILED:
+			return {
+				error: action.error.error,
+				status: action.error.status,
+				statusCode: action.error.statusCode,
+				name: action.error.name,
+				message: action.error.message,
+			};
+	}
+	return state;
+};
+
+const isFetchingNewUser: Reducer<
```

Comment:
isFetchingSite

Reply:
I haven't quite worked out how `isResolving` works, and whether we need to pass in the last known params, e.g., https://github.com/Automattic/wp-calypso/blob/d4ac5e40fdeb6498ef846399ec4004010f20d1ff/packages/data-stores/src/domain-suggestions/selectors.ts#L36 

Maybe @sirreal could school us on that magic

Reply:
That's the gist as I recall. You pass the store key, the resolver name, and an array of the arguments you passed to the resolver. More than the last known params, consider that the same resolver may be resolving different requests, or the same resolver may be resolving the same request multiple times. The params serve to the same resolvers differentiate between the same and different resolution.